### DISCUSSION TITLE
iOS 源代码分析----Masonry · /ios-yuan-dai-ma-fen-xi-masonry

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#495] [Announcements]
https://draveness.me/ios-yuan-dai-ma-fen-xi-masonry

**Comment by peaktangf** [comment]
作者还没有指明为什么需要MASCompositeConstraint类，感觉所有的约束直接存在maker中也是没问题的，都由maker代理执行约束添加的方法。个人感觉MASCompositeConstraint有点多余。
